pkg/webui/console/containers/gateway-onboarding-form/index.js

```diff
@@ -86,37 +92,64 @@ const GatewayOnboardingForm = props => {
       const gatewayId = cleanValues.ids.gateway_id
 
       try {
-        await dispatch(attachPromise(createGateway(ownerId, cleanValues, isUserOwner)))
         if (_create_api_key_cups) {
-          await generateCUPSApiKey(gatewayId)
+          const generatedCupsKey = await generateCUPSApiKey(gatewayId)
+          setCupsKey(generatedCupsKey)
         }
         if (_create_api_key_lns) {
-          await generateLNSApiKey(gatewayId)
+          const generatedLnsKey = await generateLNSApiKey(gatewayId)
+          cleanValues.lbs_lns_secret = { value: generatedLnsKey }
+          setLnsKey(generatedLnsKey)
         }
+        await dispatch(attachPromise(createGateway(ownerId, cleanValues, isUserOwner)))
         onSuccess(gatewayId)
+        setModalVisible(cupsKey || lnsKey)
```

Comment:
If you call `onSuccess(gatewayId)` here, then the site will already redirect and the modal will not be shown. You have to defer the success handler to after the modal has been resolved (that is, only when the modal is shown).